flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobGraph.java

```diff
@@ -628,4 +632,13 @@ public void writeUserArtifactEntriesToConfiguration() {
                     userArtifact.getKey(), userArtifact.getValue(), jobConfiguration);
         }
     }
+
+    public void setJobStatusHooks(List<JobStatusHook> hooks) {
```

Comment:
I think here two implementation way:
1. follow `addJars` method, method rename to `addJobStatusHooks `,  add the object of list to existing list. Maybe we can register hooks multiply instead of clear it before set.
2. follow `setClasspaths` method, jobStatusHooks initial value is `Collection.emptyList()`, here assign list to it.